omod/src/test/java/org/openmrs/module/registrationapp/form/RegisterPatientFormBuilderTest.java

```diff
@@ -41,4 +50,36 @@ public void buildFormStructure_shouldFlattenTheWidgetConfig() throws Exception {
 		assertEquals(60, fc.get("size"));
 		assertEquals(true, fc.get("resizeable"));
 	}
+	
+	/**
+	 * @see RegisterPatientFormBuilder#buildFormStructure(AppDescriptor,Boolean)
+	 * @verifies combbining of config sections
+	 */
+	@Test
+	public void buildFormStructure_shouldCombineConfigSectionsIntoOne() throws Exception {
+		// setup
+		ObjectMapper mapper = new ObjectMapper();
+		List<AppDescriptor> appDescriptors = mapper.readValue(getClass().getClassLoader().getResourceAsStream("registration_app.json"), new TypeReference<List<AppDescriptor>>() {});
+		AppDescriptor appDescriptor = appDescriptors.get(0);
+		List<String> questionIds = new ArrayList<String>();
+		questionIds.add("personAddress");
+		questionIds.add("phoneNumber");
+		questionIds.add("fathersDetails");
+		questionIds.add("mothersDetails");
+		questionIds.add("insuranceDetails");
+		questionIds.add("triageVitals");
```

Comment:
Why don't you get all the available questions out of the app descriptor?

Reply:
There is no easy way iterating over collection of `Question`s since they are not comparable.

Reply:
@Ruhanga what about?
```java
List<String> questionIds = new ArrayList<String>();
{
  Iterator<JsonNode> sectionIt = appDescriptor.getConfig().get("sections").getElements();
  while(sectionIt.hasNext()) {
    Iterator<JsonNode> questionIt = sectionIt.next().get("questions").getElements();
    while(questionIt.hasNext()) {
      questionIds.add(questionIt.next().get("id").getTextValue());
    }
  }
}
```

The point is to not decouple your unit test from the test data it replays with.